### ISSUE TITLE
[java] CloseResource false positive with late assignment of variable

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1967]
**Affects PMD Version:** 6.16.0

**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**
If the local variable holding a stream is not directly initialized but the value is assigned later on, this assigned is considered problematic. But it isn't, since it only assigns a value.

It should be no problem, if the stream is then later on used with a try-with-resources, and there are no other statements before the try-with-resources.

**Code Sample demonstrating the issue:**

``` java
import java.io.*;

public class CloseResourceFP {
    public void check(File outputFile) {
        final OutputStream os;
        if (outputFile == null) {
            os = System.out;
        } else if (outputFile.isAbsolute()) {
            os = Files.newOutputStream(outputFile.toPath());
        } else {
            os = Files.newOutputStream(new File(getProject().getBaseDir(), outputFile.toString()).toPath());
        }

        try (Writer writer = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"))) {
            renderer.render(cpd.getMatches(), writer);
        }
    }
}
```
